pkgs/by-name/at/atlauncher/package.nix

```diff
@@ -1,52 +1,66 @@
-{ copyDesktopItems, fetchurl, jre, lib, makeDesktopItem, makeWrapper, stdenv, udev, xorg }:
+{ copyDesktopItems
+, fetchFromGitHub
+, fetchurl
+, lib
+, makeDesktopItem
+, makeWrapper
+, stdenvNoCC
 
-stdenv.mkDerivation (finalAttrs: {
+, # dependencies
+  jre
+, udev
+, xorg
+}:
+
+let
   pname = "atlauncher";
-  version = "3.4.35.4";
+  version = "3.4.35.9";
 
-  src = fetchurl {
-    url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${finalAttrs.version}/ATLauncher-${finalAttrs.version}.jar";
-    hash = "sha256-M8ygN70yizJM6VEffBh/lH/DneKAzQ5UFzc3g51dja0=";
+  srcs = {
+    source = (fetchFromGitHub {
+      owner = "ATLauncher";
+      repo = "ATLauncher";
+      rev = "v${version}";
+      hash = "sha256-QbSCerrALrbHMWuPgJEthUrcYlaDEWuU6Kzwqe2mr8Q=";
+    });
+    jar = (fetchurl {
+      url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${version}/ATLauncher-${version}.jar";
+      hash = "sha256-Y2MGhzq4IbtjEG+CER+FWU8CY+hn5ehjMOcP02zIsR4=";
+    });
   };
 
-  env.ICON = fetchurl {
-    url = "https://atlauncher.com/assets/images/logo.svg";
-    hash = "sha256-XoqpsgLmkpa2SdjZvPkgg6BUJulIBIeu6mBsJJCixfo=";
-  };
+  packagingDir = "${srcs.source}/packaging/linux/_common";
+in
+stdenvNoCC.mkDerivation {
+  inherit pname version;
 
+  src = srcs.jar;
   dontUnpack = true;
 
   nativeBuildInputs = [ copyDesktopItems makeWrapper ];
 
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/bin $out/share/java
-    cp $src $out/share/java/ATLauncher.jar
+    install -D -m 444 $src $out/share/java/ATLauncher.jar
 
     makeWrapper ${jre}/bin/java $out/bin/atlauncher \
```

Comment:
Since it's available on Darwin, shouldn't it be better to use `makeBinaryWrapper`?

Reply:
Unsure how that would look like since the binary is not yet generated and isn't the required first argument of `makeWrapper` a binary?

Reply:
Just replace `makeWrapper` by `makeBinaryWrapper`, nothing else to do.

Reply:
Doesn't seem to work. I get the following error when testing the binary:
```
java.lang.NullPointerException: Cannot invoke "java.nio.file.Path.getFileSystem()" because "path" is null
```

Reply:
`makeBinaryWrapper` doesn't support shell expansion as it's used here:
```sh
--add-flags "--working-dir \"\''${XDG_DATA_HOME:-\$HOME/.local/share}/ATLauncher\"" \
```

Reply:
So, is there an alternative for this?

Reply:
I'd say leave this as is (using `makeWrapper`). `makeBinaryWrapper` would break this flag